src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -554,6 +554,15 @@ static bool mergeAttrInfo
     }
   }
 
+  if (!attrValueChanges(attr, caP, forcedUpdate, apiVersion) && !equalMetadata(md, mdNew))
+  {
+    onlyMetadataChange = true;
+  }
+  else
+  {
+    onlyMetadataChange = false;
+  }
```

Comment:
It seems you are using the global variable `onlyMetadataChange` that should be removed (see my previous comment), so I'm afraid this strategy probably is not valid.

Maybe you could "mark" showhow that the entity has modified its metadata and its value independently (note that currently the returned `actualUpdate` doesn't allow to distinguish that) and use that information in addTriggeredSubscriptions_noCache() and addTriggeredSubscriptions_withCache(), in which moment you have the `onlyMetadataChange ` field coming from the subscription and you can make a decision upon sending notification or not.

If you find that difficult I could try to provide some help (the most "heavy" part of the work, about CRUD management of the new attribute and changes in the .test files is already done).